### PR TITLE
Fixed fatal errors with PHP 7.2 and PHPUnit 7.2

### PR DESCRIPTION
**Pull request by SilverFire**
This pull request fixed methods declaration incompatibilites that lead to Fatal error with PHP 7.2 and PHPUnit 7.2

### CODE REVIEW

**Comment by Naktibalda** [comment]
Please make it compatible with PHP 5.6

File: src/Codeception/Test/Unit.php [hunk 1]
```diff
@@ -130,7 +130,7 @@ public function getReportFields()
         ];
     }
 
-    public function getDependencies()
```
Comment (Naktibalda) [inline src/Codeception/Test/Unit.php]: getDependencies is a more interesting case. Conflict wasn't cause by a change of signature.
Codeception had this method first, PhpUnit introduced it in 7.2.0, so fixing signature could cause interesting bugs.
https://github.com/sebastianbergmann/phpunit/commit/66a0765da30140eb9618b50cbeddd6223eccdf4a#diff-511c90242d5589e941582006e0485184R964

Comment (ddinchev) [reply src/Codeception/Test/Unit.php]: One solution could be to constrain codeception's phpunit version up to 7.1 until we drop the support of php `5.x`.

**Comment by Naktibalda** [comment]
@DavertMik tried to fix phpunit signature compatibility issues by moving PhpUnit related code to [phpunit-wrapper](https://github.com/Codeception/phpunit-wrapper/).

It looks like we have to move test format classes to phpunit-wrapper too.
Version without result types goes to 6.0 branch, version with result types - to 7.0 branch.

Oh great, this change will make phpunit-wrapper incompatible with older versions of Codeception, so major releases will be necessary (8.0 and 9.0? 🤣 ).

**Comment by Naktibalda** [comment]
Related issue: #5092

**Comment by Slamdunk** [comment]
I think a BC break release isn't necessary: #5097

**Pull request closed without merging**